draft-ietf-quic-qpack.md

```diff
@@ -825,22 +825,32 @@ The decoder reconstructs the Largest Reference using the following algorithm:
 
 ~~~
    if LargestReference > 0:
+     if LargestReference > 2 * MaxEntries:
+        Error
+
       LargestReference -= 1
       CurrentWrapped = TotalNumberOfInserts mod (2 * MaxEntries)
 
       if CurrentWrapped >= LargestReference + MaxEntries:
          # Largest Reference wrapped around 1 extra time
          LargestReference += 2 * MaxEntries
-      else if CurrentWrapped + MaxEntries < LargestReference
+      else if CurrentWrapped + MaxEntries < LargestReference:
          # Decoder wrapped around 1 extra time
          CurrentWrapped += 2 * MaxEntries
 
-      LargestReference += TotalNumberOfInserts - CurrentWrapped
+      LargestReference += TotalNumberOfInserts
+
+      if LargestReference <= CurrentWrapped:
```

Comment:
I put the error checking here because I am using unsigned integers.  If the majority of implementors are expected to use signed integers (after all, 2**62 required by the spec can comfortably be represented using 64 bit signed integers), then the original `LargestReference += TotalNumberOfInserts - CurrentWrapped` line can be kept and the error checking woul look like:

```
   if LargestReference <= 0:
      Error
```